### ISSUE TITLE
Validator does not catch constraint

### ISSUE THREAD

**Issue by ewoutkramer** [FirelyTeam/firely-validator-api#265]
(Issue moved over from Jira, original poster: Marten)

Nictiz defines two slice on Observation.coding in this profile 
https://simplifier.net/packages/nictiz.fhir.nl.stu3.zib2017/2.2.13/files/2039136/~overview 

There’s a comment on the second slice stating:
"This slice binds the ValueSet from the HCIM, which overlaps the ValueSet from the core FHIR Observation. To satisfy the FHIR profiling rules, the code from the core FHIR ValueSet SHOULD be included as well if a code from this ValueSet is included."

And an invariant on interpretation.coding checking the same:
coding.where(system = ‘http://snomed.info/sct' and (code = '281302008' or '281300000' or '11896004' or '30714006' or '131196009') ).exists() implies coding.where(system = 'http://hl7.org/fhir/v2/0078').exists()

However, the validator does not throw an error when only one of the SNOMED codes is present. 
This is either an error in the invariant (which I don’t see), or an issue in the validator.

Example JSON that should throw an error:

```
{
    "messageSchemaVersion": "2021-05-25",
    "messageId": "91",
    "single": {
        "resource": {
            "resourceType": "Observation",
            "id": "LABDCR1002171736799209",
            "meta": {
                "profile": [
                    "http://nictiz.nl/fhir/StructureDefinition/zib-LaboratoryTestResult-Observation"
                ],
                "tag": [
                    {
                        "system": "http://mumc.nl/fhir/schema/Source/v1.0/source"
                    }
                ]
            },
            "status": "final",
            "_status": {
                "extension": [
                    {
                        "url": "http://nictiz.nl/fhir/StructureDefinition/code-specification",
                        "valueCodeableConcept": {
                            "coding": [
                                {
                                    "system": "urn:oid:2.16.840.1.113883.2.4.3.11.60.40.4.16.1",
                                    "code": "final",
                                    "display": "Definitief"
                                }
                            ]
                        }
                    }
                ]
            },
            "category": [
                {
                    "coding": [
                        {
                            "system": "http://snomed.info/sct",
                            "code": "49581000146104"
                        }
                    ],
                    "text": "Laboratory test finding (finding)"
                }
            ],
            "code": {
                "coding": [
                    {
                        "system": "http://loinc.org",
                        "code": "62238-1"
                    }
                ],
                "text": "Glomerular filtration rate/1.73 sq M.predicted"
            },
            "subject": {
                "reference": "Patient/0041000000"
            },
            "valueString": "55.4 mL/min/1.73m",
            "interpretation": {
                "coding": [
                    {
                        "system": "http://snomed.info/sct",
                        "code": "281300000",
                        "display": "Onder referentiewaarde"
                    }
                ]
            },
            "specimen": {
                "reference": "Specimen/LABDCR100217173679"
            },
            "referenceRange": [
                {
                    "low": {
                        "value": "90",
                        "unit": "mL/min/1.73m",
                        "system": "http://unitsofmeasure.org",
                        "code": "mL/min/1.73m"
                    }
                }
            ]
        }
    }
}
```

**Comment by ewoutkramer** [comment]
The invariant is incorrect:

`coding.where(system = ‘http://snomed.info/sct' and (code = '281302008' or '281300000' or '11896004' or '30714006' or '131196009') ).exists() implies coding.where(system = 'http://hl7.org/fhir/v2/0078').exists()`

should be

coding.where(system = ‘http://snomed.info/sct' and (code = '281302008' or **code=**'281300000' or **code=**'11896004' or **code=**'30714006' or **code=**'131196009') ).exists() implies coding.where(system = 'http://hl7.org/fhir/v2/0078').exists()

@mmsmits - do we need to signal someone about this?

**Comment by mmsmits** [comment]
Reported:
https://nictiz.atlassian.net/browse/ZIBFHIR-63
